validator_client/http_api/src/tests.rs

```diff
@@ -46,43 +46,39 @@ struct ApiTester {
     validator_store: Arc<ValidatorStore<TestingSlotClock, E>>,
     url: SensitiveUrl,
     slot_clock: TestingSlotClock,
-    _validator_dir: TempDir,
     _test_runtime: TestRuntime,
 }
 
 impl ApiTester {
     pub async fn new() -> Self {
-        let mut config = Config::default();
+        let mut config = ValidatorStoreConfig::default();
         config.fee_recipient = Some(TEST_DEFAULT_FEE_RECIPIENT);
         Self::new_with_config(config).await
     }
 
-    pub async fn new_with_config(mut config: Config) -> Self {
+    pub async fn new_with_config(config: ValidatorStoreConfig) -> Self {
         let log = test_logger();
 
-        let validator_dir = tempdir().unwrap();
-        let secrets_dir = tempdir().unwrap();
+        let validator_dir: PathBuf = tempdir().unwrap().path().into();
+        let secrets_dir: PathBuf = tempdir().unwrap().path().into();
```

Comment:
I don't think this is a good idea, as the tempdir will get dropped and deleted?

I'm surprised the tests are passing currently